pkgs/by-name/ap/aphorme/package.nix

```diff
@@ -0,0 +1,67 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, fontconfig
+, pkg-config
+, wayland
+, libxkbcommon
+, makeWrapper
+, libGL
+, xorg
+, testers
+, aphorme
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "aphorme";
+  version = "0.1.19";
+
+  src = fetchFromGitHub {
+    owner = "AnyTimeTraveler";
+    repo = "aphorme_launcher";
+    rev = "v${version}";
+    hash = "sha256-WcZOS/iM2FUlOKLeUUJx6sDIU/sK38EeuyucxlLIIKw=";
+  };
+
+  cargoHash = "sha256-OJbG+D8bxfMeExxX9xlhWDReT4Y6hEixddn6ep7dAPs=";
+
+  # No tests exist
+  doCheck = false;
+
+  buildInputs = [
+    fontconfig
+    libxkbcommon
+    libGL
+
+    # WINIT_UNIX_BACKEND=wayland
+    wayland
+
+    # WINIT_UNIX_BACKEND=x11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libX11
+  ];
+  nativeBuildInputs = [ makeWrapper pkg-config ];
+
+  postFixup = ''
+    wrapProgram $out/bin/aphorme \
+      --suffix LD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs}
```

Comment:
This is strongly discouraged. Please try runtimeDependencies with autoPatchelfHook.
[nixos.org/manual/nixpkgs/unstable/#setup-hook-autopatchelfhook](https://nixos.org/manual/nixpkgs/unstable/#setup-hook-autopatchelfhook)

Also not all of buildInputs are needed in LD_LIBRARY_PATH?

Reply:
That took a bit to figure out, but I think I've solved it to satisfaction.
The documentation on the nixpkgs manual could use an example for this mechanism.